### ISSUE TITLE
3.2.0: SmartLock Email picker always shown

### ISSUE THREAD

**Issue by porkyhead** [firebase/FirebaseUI-Android#1118]
* Android device: Google Pixel
  * Android OS version: 8.1
  * Google Play Services version: 11.8.0
  * Firebase/Play Services SDK version: 11.8.0
  * FirebaseUI version: 3.2.0
  
After upgrading to version 3.2.0 Smartlock email picker is shown even though authentcation is set to phone or Facebook. After email picker has been closed, the correct flow follows.

  ```
       startActivityForResult(
                AuthUI.getInstance()
                        .createSignInIntentBuilder()
                        .setLogo(R.drawable.ic_happenings)
                        .setPrivacyPolicyUrl(getString(R.string.privacy_policy_link))
                        .setAvailableProviders(
                                Arrays.asList(
                                        new AuthUI.IdpConfig.PhoneBuilder().build()
                                ))
                        .build(),
                RC_SIGN_IN);
  ```

**Comment by samtstern** [comment]
@porkyhead can you provide a little more detail?  Do you mean this picker?
![smartlock](https://user-images.githubusercontent.com/8466666/35286000-9a2281c4-0013-11e8-9df4-15951861e3f7.png)

That is shown when you have previously saved an account with FirebaseUI and allows you to log in with a single tap.  It's the intended behavior for now.

Or do you mean something else?

**Comment by RuiVarela** [comment]
upgrading to 3.2.0 is hell of a ride. that behavior changed out of nowhere...
I currently disabled smartlock
 .setIsSmartLockEnabled(false), 

to set the same behavior as before, you need to clear your app data

**Comment by samtstern** [comment]
@RuiVarela can you help explain what changed exactly?

**Comment by RuiVarela** [comment]
sure, 

if you use this snipped
```
        m_activity.startActivityForResult(
                AuthUI.getInstance()
                        .createSignInIntentBuilder()
                        .setAvailableProviders(Arrays.asList(new AuthUI.IdpConfig.PhoneBuilder().build()))
                        .build(),
                REQUEST_LOGIN_PHONE);
```
which was what we used, and if you have a picker with "email, phone, facebook, etc..." we were expecting that by using the phone you would be directed to phone validation screen,
but now it pops up a smart lock thing, which is surprising for the user and that eventual crashes the app

![01](https://user-images.githubusercontent.com/11543973/35290916-984b2766-0063-11e8-90a0-49f46733a064.png)
![02](https://user-images.githubusercontent.com/11543973/35290917-9971a6f6-0063-11e8-81ce-3b21e0458121.png)
![03](https://user-images.githubusercontent.com/11543973/35290922-9b652708-0063-11e8-9b2a-95099874b77f.png)

```
01-23 17:31:19.617 11831-11831/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: co.triller.droid, PID: 11831
                                                   java.lang.IllegalStateException: Provider password couldn't not be found in [IdpConfig{mProviderId='phone', mParams=Bundle[EMPTY_PARCEL]}]
                                                       at com.firebase.ui.auth.util.a.b.a(ProviderUtils.java:104)
                                                       at com.firebase.ui.auth.ui.email.EmailActivity.c(EmailActivity.java:123)
                                                       at com.firebase.ui.auth.ui.email.a$2.a(CheckEmailFragment.java:203)
                                                       at com.firebase.ui.auth.ui.email.a$2.onSuccess(CheckEmailFragment.java:199)
                                                       at com.google.android.gms.tasks.zzj.run(Unknown Source)
                                                       at android.os.Handler.handleCallback(Handler.java:751)
                                                       at android.os.Handler.dispatchMessage(Handler.java:95)
                                                       at android.os.Looper.loop(Looper.java:154)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6077)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
```

to keep the old behaviour, or not showing up the "User accounts list" I disabled the smart lock

**Comment by RuiVarela** [comment]
the expression "Continue With" is really not explanatory of what is happening, I just thought that I was going through and email validation process

**Comment by samtstern** [comment]
@RuiVarela ah ok thanks, I see what's happening here.  At some point you must have enabled the email flow (maybe in testing) and gone far enough to save an email credential.

If in your production app you've only ever called `setAvailableProviders()` with the phone config then your users will never see this error, but we could be more defensive anyway.

I'll see what I can do.

**Comment by samtstern** [comment]
Also to be clear, FirebaseUI no control over the "continue with" messaging or the styling of that email dialog, that comes directly from SmartLock.

**Comment by SUPERCILEX** [comment]
@samtstern sorry for the late reply, I'm pretty sure it doesn't have to do with providers but instead the new credential client. We have to redo something like https://github.com/firebase/FirebaseUI-Android/pull/708. I'm seeing the same behavior as people here in that all my emails and Google accounts are showing up at the beginning.

**Comment by samtstern** [comment]
@SUPERCILEX got it, I am digging into this now.

**Comment by SUPERCILEX** [comment]
Awesome!

**Comment by samtstern** [comment]
@SUPERCILEX I see that we always set `setPasswordLoginSupported` to `true`.  Were we saving phone number credentials as password credentials or something?

**Comment by SUPERCILEX** [comment]
@samtstern Not yet, but I did add support to save phone numbers in #962 and plan on bringing that over at some point. Currently tracked down the issue to https://github.com/firebase/FirebaseUI-Android/commit/ff16d0dd4d1e51c81e50cf9c4139e0a919ab6a4e#diff-741502abb522ecd56e7c30231e015bdbL138, investigating.

**Comment by SUPERCILEX** [comment]
Bingo! PR in a sec.

**Comment by samtstern** [comment]
@SUPERCILEX nice!  Btw the reason we don't save phone numbers is because SmartLock does a pretty bad job of presenting them properly in the UI.  So I'm tracking an internal bug to improve that.

**Comment by SUPERCILEX** [comment]
@samtstern Oh sweet! I do remember having to hack it in, glad to know improvements are coming! 😄

**Comment by samtstern** [comment]
This has been fixed and released in `3.2.1`.
